scripts/lib/eslint-plugin.ts

```diff
@@ -295,4 +295,137 @@ exports.rules = {
 		},
 		defaultOptions: [],
 	}),
+	'tagged-components': ESLintUtils.RuleCreator.withoutDocs({
+		create(context) {
+			function isComponentName(node: TSESTree.Node) {
+				return node.type === 'Identifier' && /^[A-Z]/.test(node.name)
+			}
+
+			function checkComponentDeclaration(
+				services: utils.ParserServices,
+				node: TSESTree.VariableDeclarator | TSESTree.FunctionDeclaration,
+				propsType: ts.TypeNode | undefined
+			) {
+				const declaration = findTopLevelParent(node)
+				const comments = context.getSourceCode().getCommentsBefore(declaration)
+
+				// we only care about components tagged as public
```

Comment:
actually, why only the `@public` ones? feels like eventually any item can eventually become `@public`. i don't feel strongly on it though and can see why we wouldn't need it for now.

Reply:
Mostly to avoid this creating churn and developer friction where it's not needed - if a component isn't public, it's not documented, so it doesn't need the `@react` tag. If it doesn't have the `@react` tag, it doesn't need to have its props specified in a very limited way